### DISCUSSION TITLE
checkboxes to toggle between different views in the same layer content

### DISCUSSION THREAD

**Discussion by Kaoutar-87** [Kitware/trame#273] [Q&A]
hello dear community,

I am currently learning VTK and Trame, and I have encountered an issue related to implementing checkbox functionality. 
Here is a brief overview of the problem:

1) I created 3 geometries : cone , cube and sphere .
2) Each object has an actor, renderer and renderer window
 3) i added a checkbox to the toolbar of the layout ,then added a function called "checkbox_clicked" to handle the checkbox click event.

The objective is that  The checkbox allows the user to toggle between different views of the visualization.

It doesn't work , and i don't know what i'm missing , can you please help solve this issue .
Thank you for your time and response.

here's a part of the code : 

```python
  # Function to handle checkbox click event

def checkbox_clicked(value):
    view = vtk_widgets.VtkLocalView(renWindow2)
    ctrl.on_server_ready.add(view.update)
    ctrl.view_update = view.update 
    ctrl.view_reset_camera = view.reset_camera 

# ---------------------------------------------------------
# View 
# ---------------------------------------------------------

with SinglePageLayout(server) as layout:
    layout.icon.click = ctrl.view_reset_camera
    layout.title.set_text("Simple Application")

    with layout.toolbar:
        vuetify.VSpacer()
        vuetify.VDivider(vertical=True, classes="mx-2")
        vuetify.VCheckbox(label="Use View2",click=checkbox_clicked)

    with layout.content:
        with vuetify.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
        ):
            view = vtk_widgets.VtkLocalView(renWindow1)
            ctrl.view_update = view.update
            ctrl.view_reset_camera = view.reset_camera
            ctrl.on_server_ready.add(view.update)
```

**Comment by jourdain** [comment]
Can you provide the full code? I'm not sure if you have several renderWindow [like here](https://github.com/Kitware/trame/discussions/269) or just one and you just toggle actor visibilities. 

One issue with the LocalView is that it does not like being empty (all actor not visible).

**Comment by jourdain** [comment]
For the multi one, you need to make sure each instance of the VtkLocalView has a unique `ref="view_xxx"`.
